## trace ancestor of hap1 hap2 of huyou
## Method 1: kpart.py https://github.com/sc-zhang/kPart
```
## input bin splited chromosomes, ancestor species reads
Mandarin: Citrus reticulata, /scratch/pawsey0399/yjia/huyou/ancestor_tracing
https://www.ncbi.nlm.nih.gov/bioproject/PRJNA388397/
https://www.ncbi.nlm.nih.gov/sra?linkname=bioproject_sra_all&from_uid=388397 ## sample info

Pummelo: Citrus maxima
https://www.nature.com/articles/ng.3839 
https://www.ncbi.nlm.nih.gov/nuccore/MKYQ00000000
https://www.ncbi.nlm.nih.gov/sra?LinkName=biosample_sra&from_uid=4901528 ## sample info
https://www.ebi.ac.uk/ena/browser/view/PRJNA318855

Citron: Citrus medica
http://citrus.hzau.edu.cn/download.php
https://www.ncbi.nlm.nih.gov/bioproject/PRJNA320023/
https://www.ncbi.nlm.nih.gov/sra?LinkName=biosample_sra&from_uid=4914485

## split chromosomes into 1Mb windows
bioawk -c fastx '{print $name "\t" length($seq)}' hap1_updated.fasta |head -n9 > hap1_updated.chrom.sizes
bedtools makewindows -g hap1_updated.chrom.sizes -w 1000000 > hap1_1Mb_bins.bed ## create bin bed file
awk '{print $0"\t"$1 NR}' hap1_1Mb_bins.bed > tmp && mv tmp hap1_1Mb_bins.bed ## add 4th name column
 hap1_1Mb_bins.bed ## have to be tab-delimited
bedtools getfasta -fi hap1_updated.fasta -bed hap1_1Mb_bins.bed -fo hap1_1Mb_bins.fasta  -name
```
## bwa align hap1 hap2 to mandarin
```
#!/bin/bash --login

#SBATCH --job-name=bwa
#SBATCH --partition=work
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=64
#SBATCH --time=24:00:00
#SBATCH --account=pawsey0399
#SBATCH --export=NONE

module load bwa/0.7.17--h7132678_9
module load samtools/1.15--h3843a85_0

#REF=./hap1_updated.fasta
REF=/scratch/pawsey0399/yjia/huyou/ancestor_tracing/split_genome/hap1_1Mb_bins.fasta

srun --export=all -n 1 -c 64 bwa index $REF
srun --export=all -n 1 -c 64 bwa mem -M -t 64 -R '@RG\tID:mandarin\tLB:mandarin\tPL:ILLUMINA\tPM:HISEQ\tSM:mandarin' $REF trimmed.SRR5837537_1.fastq.gz trimmed.SRR5837537_2.fastq.gz |samtools view -@ 12 -Sb - | samtools sort -@ 12 -o hap1_split_SRR5837537.bam
srun --export=all -n 1 -c 64 samtools index -@ 64 hap1_split_SRR5837537.bam
```
## bwa align hap1 hap2 to pummelo
```
#!/bin/bash --login

#SBATCH --job-name=bwa
#SBATCH --partition=work
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=64
#SBATCH --time=24:00:00
#SBATCH --account=pawsey0399
#SBATCH --export=NONE

module load bwa/0.7.17--h7132678_9
module load samtools/1.15--h3843a85_0

#REF=./hap1_updated.fasta
REF=/scratch/pawsey0399/yjia/huyou/ancestor_tracing/split_genome/hap1_1Mb_bins.fasta

srun --export=all -n 1 -c 64 bwa index $REF
srun --export=all -n 1 -c 64 bwa mem -M -t 64 -R '@RG\tID:mandarin\tLB:mandarin\tPL:ILLUMINA\tPM:HISEQ\tSM:mandarin' $REF trimmed.SRR4294658_1.fastq.gz trimmed.SRR4294658_2.fastq.gz |samtools view -@ 12 -Sb - | samtools sort -@ 12 -o hap1_split_SRR4294658.bam
srun --export=all -n 1 -c 64 samtools index -@ 64 hap1_split_SRR4294658.bam
```
## run kPart for hap1 and hap2
```
mamba install bioconda::pysam
cd /scratch/pawsey0399/yjia/tools
git clone https://github.com/sc-zhang/kPart.git
```

```
#!/bin/bash --login

#SBATCH --job-name=kpart1
#SBATCH --partition=work
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=64
#SBATCH --time=12:00:00
#SBATCH --account=pawsey0399
#SBATCH --export=NONE

conda activate base
KPART=/scratch/pawsey0399/yjia/tools/kPart/kPart.py
REF=/scratch/pawsey0399/yjia/huyou/ancestor_tracing/split_genome/hap1_1Mb_bins.fasta
srun --export=all -n 1 -c 64 $KPART -r $REF -b ./ -k 21 -t 64
```
## output interpretation
```
## ancestor read data
mandarin: SRR5837537
pummelo: SRR4294658
citron: SRR4010249

## sed_species
cat species |while IFS=':' read R1 R2;do echo "s/hap1_split_"$R2"/"$R1"/";done > sed_species
s/hap1_split_SRR5837537/mandarin/
s/hap1_split_SRR4294658/pummelo/
s/hap1_split_SRR4010249/citron/

s/hap2_split_SRR5837537/mandarin/
s/hap2_split_SRR4294658/pummelo/
s/hap2_split_SRR4010249/citron/

sed -f sed_species kmer_dist_classified.csv > seded_kmer_dist_classified.csv

##count origins
awk -F ',' 'NR>1{print $NF}' seded_kmer_dist_classified.csv |sort|uniq -c

##hap2
      1 mandarin
    282 pummelo
     26 Undetermined
##hap1
    276 mandarin
     24 pummelo
      4 Undetermined

## create bed file for circos color visualization

```
