## process RNAseq from published paper
### working directory: /scratch/pawsey0399/yjia/huyou/rnaSeq
```
## input data: /scratch/pawsey0399/yjia/huyou/genome_annotation/gemoma/hap2updated_less
mamba install bioconda::gffread
gffread -w transcripts.fa -g ../hap1_updated.softmasked.fasta final_annotation.gff -F
gffread -w transcripts.fa -g ../hap2_updated.softmasked.fasta final_annotation.gff -F

## clean and modify transcripts sequence
/scratch/pawsey0399/yjia/huyou/rnaSeq
ln -s /scratch/pawsey0399/yjia/huyou/genome_annotation/gemoma/hap1updated_less/hap1updated_less_transcripts.fa
ln -s /scratch/pawsey0399/yjia/huyou/genome_annotation/gemoma/hap2updated_less/hap2updated_less_transcripts.fa

awk '/^>/ {for (i=1; i<=NF; i++) if ($i ~ /Name=/) print ">"$i; next} {print}' hap1updated_less_transcripts.fa > hap1updated_less_transcripts.fasta
awk '/^>/ {for (i=1; i<=NF; i++) if ($i ~ /Name=/) print ">"$i; next} {print}' hap2updated_less_transcripts.fa > hap2updated_less_transcripts.fasta

sed -i 's/Name=//' hap1updated_less_transcripts.fasta
sed -i 's/Name=//' hap2updated_less_transcripts.fasta

cut -d ';' -f1 hap1updated_less_transcripts.fasta > tmp && mv tmp hap1updated_less_transcripts.fasta
cut -d ';' -f1 hap2updated_less_transcripts.fasta > tmp && mv tmp hap2updated_less_transcripts.fasta

sed -i 's/^>HY/>H1Y/' hap1updated_less_transcripts.fasta
sed -i 's/^>HY/>H2Y/' hap2updated_less_transcripts.fasta

cat hap1updated_less_transcripts.fasta hap2updated_less_transcripts.fasta > merged_transcripts.fa
```
### run
```
ls --color=never *fastq.gz|paste - - > sample_pairs
## fastp.sh
while read R1 R2;
do
	fastp \
-i ./$R1 \
-I ./$R2 \
-o ./trimmed.$R1 \
-O ./trimmed.$R2 \
--qualified_quality_phred 20 \
--cut_front \
--cut_tail \
--length_required 30 \
--detect_adapter_for_pe \
--thread 64
done < sample_pairs

## kallisto.sh
##build index
kallisto index -i merged_transcripts.idx merged_transcripts.fa

## quantification
ls 'trimmed*fq.gz' | paste - - | while read R1 R2;
do
	D=$(echo $R1 | cut -d '.' -f2)
	kallisto \
	quant \
	-i merged_transcripts.idx \
	-t 64 \
	-b 100 \
	-o $D \
	$R1 \
	$R2
done

## rnaseq.conf
conda activate base
srun --export=all -n 1 -c 64  bash fastp.sh &> log1.txt
srun --export=all -n 1 -c 64  bash kallisto.sh &> log2.txt
```
