## process RNAseq from published paper
### working directory: /scratch/pawsey0399/yjia/huyou/rnaSeq
```
## input data: /scratch/pawsey0399/yjia/huyou/genome_annotation/gemoma/hap2updated_less
mamba install bioconda::gffread
gffread -w transcripts.fa -g ../hap1_updated.softmasked.fasta final_annotation.gff -F
gffread -w transcripts.fa -g ../hap2_updated.softmasked.fasta final_annotation.gff -F

## clean and modify transcripts sequence
/scratch/pawsey0399/yjia/huyou/rnaSeq
ln -s /scratch/pawsey0399/yjia/huyou/genome_annotation/gemoma/hap1updated_less/hap1updated_less_transcripts.fa
ln -s /scratch/pawsey0399/yjia/huyou/genome_annotation/gemoma/hap2updated_less/hap2updated_less_transcripts.fa

awk '/^>/ {for (i=1; i<=NF; i++) if ($i ~ /Name=/) print ">"$i; next} {print}' hap1updated_less_transcripts.fa > hap1updated_less_transcripts.fasta
awk '/^>/ {for (i=1; i<=NF; i++) if ($i ~ /Name=/) print ">"$i; next} {print}' hap2updated_less_transcripts.fa > hap2updated_less_transcripts.fasta

sed -i 's/Name=//' hap1updated_less_transcripts.fasta
sed -i 's/Name=//' hap2updated_less_transcripts.fasta

cut -d ';' -f1 hap1updated_less_transcripts.fasta > tmp && mv tmp hap1updated_less_transcripts.fasta
cut -d ';' -f1 hap2updated_less_transcripts.fasta > tmp && mv tmp hap2updated_less_transcripts.fasta

sed -i 's/^>HY/>H1Y/' hap1updated_less_transcripts.fasta
sed -i 's/^>HY/>H2Y/' hap2updated_less_transcripts.fasta

cat hap1updated_less_transcripts.fasta hap2updated_less_transcripts.fasta > merged_transcripts.fa
```
